webanno-api-dao/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/dao/AnnotationSchemaServiceImpl.java

```diff
@@ -961,9 +961,9 @@ public boolean upgradeCasIfRequired(Iterable<CAS> aCasIter, Project aProject)
         // Check if the current CAS already contains the required type system
         boolean upgradePerformed = false;
         for (CAS cas : aCasIter) {
-            CasStorageSession.get().assertWritingPermitted(cas);
             
             if (cas != null && isUpgradeRequired(cas, ts)) {
+                CasStorageSession.get().assertWritingPermitted(cas);
```

Comment:
I don't think we should check if writing is permitted only if we actually detect that an update is necessary. Even if no upgrade is necessary that the system is in principle allowed to perform the write operation. Was moving this assert into the condition necessary to fix the issue?

Reply:
Yes, the cas was null at this point.

Reply:
I'll separate the null check from the upgrade check. Null CASes should just be skipped.